docs/DafnyRef/Grammar.md

```diff
@@ -561,7 +561,40 @@ A ``FormalsOptionalIds`` is a formal parameter list in which the types are requi
 but the names of the parameters are optional. This is used in algebraic
 datatype definitions.
 
-### 2.6.5. Numeric Literals
+### 2.6.5. Quantifier Domains {#sec-quantifier-domains}
+````grammar
+QuantifierDomain(allowLemma, allowLambda) =
+    QuantifierVarDecl(allowLemma, allowLambda) 
+    { "," QuantifierVarDecl(allowLemma, allowLambda) }
+
+QuantifierVarDecl(allowLemma, allowLambda) =
+    IdentTypeOptional
+    [ <- Expression(allowLemma, allowLambda) ]
+    { Attribute }
+    [ | Expression(allowLemma, allowLambda) ]
+````
```

Comment:
Only three backticks are necessary.

Also, I think we all agreed on that and all the documentation needs to be reformatted accordingly, but could you perhaps start the effort by moving this grammar to the end of this section and replace it with a few examples? That would be less frightening.

Reply:
> Are you suggesting moving it to the end of Section 2, as a 2.7?

No I'm not suggesting changing numbering or the order of sections and sub-sections.
I'm only suggesting to not place the grammar at the beginning of this section, only at the end (until it will be moved to another place). The beginning of the section should consist of introductory text to the features and example covering most use cases.

So, instead of:
```
### 2.6.5. Quantifier Domains {#sec-quantifier-domains}
<frightening and unreadable grammar definitions>
<introduction text>
<examples>
<more text>
```

I strongly suggest you write:
```
### 2.6.5. Quantifier Domains {#sec-quantifier-domains}
<introduction text>
<examples>
<more text>
<frightening and unreadable grammar definitions>
```

Reply:
Aha thanks for clarifying. I will gladly start setting a better example. :)